app/merginapi.cpp

```diff
@@ -739,21 +737,22 @@ void MerginApi::authorizeFinished()
   if ( r->error() == QNetworkReply::NoError )
   {
     InputUtils::log( "auth", QStringLiteral( "Success" ) );
-    QJsonDocument doc = QJsonDocument::fromJson( r->readAll() );
+    const QByteArray data = r->readAll();
+    QJsonDocument doc = QJsonDocument::fromJson( data );
     if ( doc.isObject() )
     {
       QJsonObject docObj = doc.object();
       mUserAuth->setFromJson( docObj );
       mUserInfo->setFromJson( docObj );
-      emit authChanged();
     }
     else
     {
-      mUserAuth->setUsername( QString() );
-      mUserAuth->setPassword( QString() );
+      whileBlocking( mUserAuth )->setUsername( QString() ); //clearTokenData emits the authChanged
+      whileBlocking( mUserAuth )->setPassword( QString() ); //clearTokenData emits the authChanged
       mUserAuth->clearTokenData();
       emit authFailed();
       InputUtils::log( "auth", QStringLiteral( "FAILED - invalid JSON response" ) );
+      qDebug() << data;
```

Comment:
do we want qDebug?

Reply:
i will fix in the following PR

Reply:
i think it is useful, since i spend significant time finding out that response from server is 301 - moved pernamently and trying to debug it all to see what going on is difficutt